bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/internal/IdentProtocol.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.irobot.internal;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.stream.JsonReader;
+
+/**
+ * iRobot discovery and identification protocol
+ *
+ * @author Pavel Fedin - Initial contribution
+ *
+ */
+public class IdentProtocol {
```

Comment:
I can't because this also applies to inner IdentData class, which is deserialized from json, and is, by your own criteria, "data only".
In fact IdentData is used as a namespace, it doesn't contain any single data field and only static functions. Does @NonNullByDefault even make sense here ?